### ISSUE TITLE
How about playing sound inbuilt in PDF?

### ISSUE THREAD

**Issue by thosch6** [mozilla/pdf.js#6078]
I have many PDF with inbuilt sound. By clicking on a graphic in Adobe Reader, it will play the wav file I linked to the musical notation. HTML5 can do sound too. Why not pdf.js? That would be awesome!

Here's a sample:
http://www.guitar.ch/pdf/en/demo/GuitarLovers-LicksRiffs-4v0-en-demo.pdf

**Comment by timvandermeij** [comment]
Requires the `/Screen` annotation to be implemented, so I'm also labeling this as annotation work.

**Comment by thosch6** [comment]
Thank you for taking notice! Are there any plans to incorporate sound playback in the near future?

**Comment by timvandermeij** [comment]
@thosch6 We first have to refactor the annotation layer in order to properly implement this. The refactoring work has already started, but is is not clear yet when it will be done.

**Comment by humphd** [comment]
Any update on this?  I might be willing to implement if there was a clear path.

**Comment by humphd** [comment]
@yurydelendik, is it possible for me to do something similar to https://github.com/mozilla/pdf.js/pull/4376 with this?  I really need audio support in PDFs on the web.

**Comment by brendandahl** [comment]
@humphd Yes, something that is preffed off by default and comes with tests would help it to be merged more quickly.

**Comment by tomorrowsj** [comment]
Is there any progress on this funtion of playing inbuilt sound or updating information?

**Comment by Snuffleupagus** [comment]
> Here's a sample:
> http://www.guitar.ch/pdf/en/demo/GuitarLovers-LicksRiffs-4v0-en-demo.pdf

Given that the file is no longer available there, nor on the Internet Archive, I'd suggest closing this issue as a duplicate of #2787 since the two features ought to be very similar.
(Also, the simplest way to "support" these Annotation-types might be to treat them basically like FileAttachment ones...)

**Comment by thosch6** [comment]
Here is the new link:
http://www.guitar.ch/pdf/en/demo/GuitarLovers-LicksRiffs-5v0-en-demo.pdf <http://www.guitar.ch/pdf/en/demo/GuitarLovers-LicksRiffs-5v0-en-demo.pdf>

And no, audio and video are not the same! It is rather annoying that pdf.js does not support audio...

**Comment by IbrahimElk** [comment]
would it be possible to get a status update on this issue ?

**Comment by jarmitage** [comment]
Can anyone with technical insight explain what would actually be required to make this work?

**Comment by calixteman** [comment]
I've a WIP patch for adding support for video/sound annotations.
It should be implemented in the next days.
